### DISCUSSION TITLE
Teleportation Delay

### DISCUSSION THREAD

**Discussion by BloodshotPico** [Zrips/CMI#4413] [Feature Requests]
Group Teleportation Delay via Permissions

---
When a user teleports they have the teleportation delay via the config setting, however I would be looking to be able to give user and or groups different teleportation delays. So when a user is trying to teleport they have to wait <x> amount of seconds before they actually get teleported. 

This would be good to have so that different groups and users don't have to wait as long if they're progressing through something like a group list etc. This would also be good for those higher ranking groups because they have to wait less time and have less time for someone to cancel their teleportation. :)

~Blood

**Comment by FatCatTuxedo** [comment]
Would also like this.

**Comment by mrfloris** [comment]
Just as information, it's already possible to have a bypass for one group, while having a warmup for another group.
```
> cmi checkperm warmup
cmi.warmupbypass.[commandname] - Allows to bypass particular CMI command warmup
cmi.command.tpahere.warmupbypass - Allows to bypass tpahere command warmup
cmi.command.tpa.warmupbypass - Allows to bypass tpa command warmup
>
```
Moving the warmup time to a permission like below would be kinda nice. That way we could also do it on a per command basis. I am not aware of the technical limitations or complications though. 
```
cmi.command.warmups.tpa.5 (default)
cmi.command.warmups.tpa.3 (vip)
cmi.command.warmups.tpa.1 (mvp)
cmi.command.warmups.tpa.0 (team)
```

**Reply by BloodshotPico** [reply]
Not entirely sure on how that actually works honestly I haven't messed too much with that kind of thing or CMI in general really. Might be doing some of that testing in the coming months here. Always wanted to get to mess around with it.
